pkg/api/handlers/mcpcatalogs.go

```diff
@@ -72,78 +74,38 @@ func (h *MCPCatalogHandler) Refresh(req api.Context) error {
 	return req.Update(&catalog)
 }
 
-// Create creates a new catalog.
-func (h *MCPCatalogHandler) Create(req api.Context) error {
+// Update updates a catalog (admin only, default catalog only).
+func (h *MCPCatalogHandler) Update(req api.Context) error {
 	var manifest types.MCPCatalogManifest
 	if err := req.Read(&manifest); err != nil {
 		return fmt.Errorf("failed to read catalog manifest: %w", err)
 	}
 
-	// Validate the URLs
-	for _, urlStr := range manifest.SourceURLs {
-		u, err := url.Parse(urlStr)
-		if err != nil {
-			return types.NewErrBadRequest("invalid URL: %v", err)
-		}
-
-		if u.Scheme != "https" {
-			return types.NewErrBadRequest("only HTTPS URLs are supported")
-		}
-	}
-
-	catalog := v1.MCPCatalog{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: system.CatalogPrefix,
-			Namespace:    req.Namespace(),
-		},
-		Spec: v1.MCPCatalogSpec{
-			DisplayName:    manifest.DisplayName,
-			SourceURLs:     manifest.SourceURLs,
-			AllowedUserIDs: manifest.AllowedUserIDs,
-		},
-	}
-
-	if err := req.Create(&catalog); err != nil {
-		return fmt.Errorf("failed to create catalog: %w", err)
-	}
-
-	return req.Write(convertMCPCatalog(catalog))
-}
-
-// Update updates a catalog.
-func (h *MCPCatalogHandler) Update(req api.Context) error {
-	var manifest types.MCPCatalogManifest
-	if err := req.Read(&manifest); err != nil {
-		return fmt.Errorf("failed to read catalog manifest: %w", err)
+	catalogID := req.PathValue("catalog_id")
+	if catalogID != system.DefaultCatalog {
+		return types.NewErrBadRequest("only the default catalog can be updated")
 	}
 
 	var catalog v1.MCPCatalog
-	if err := req.Get(&catalog, req.PathValue("catalog_id")); err != nil {
+	if err := req.Get(&catalog, catalogID); err != nil {
 		return fmt.Errorf("failed to get catalog: %w", err)
 	}
 
-	if catalog.Spec.IsReadOnly {
-		return types.NewErrBadRequest("cannot update a read-only catalog")
-	}
-
-	if manifest.DisplayName == "" {
-		return types.NewErrBadRequest("display name is required")
-	}
-
+	// The only field that can be updated is the source URLs.
 	for _, urlStr := range manifest.SourceURLs {
-		u, err := url.Parse(urlStr)
-		if err != nil {
-			return types.NewErrBadRequest("invalid URL: %v", err)
-		}
+		if urlStr != "" && urlStr != h.defaultCatalogPath {
```

Comment:
nit: This fixes the specific issue I pointed out, but what if I want to do some local testing of catalogs?

One suggestion: make the check below `if u.Host != "" && u.Host != "localhost" && u.Scheme != "https" {`.

This is a nit because I am not sure if this is something we'll want to do, and the fix would be fairly straightforward.

Reply:
I think I will hold off on this for now, until we get direction that this is something that we want to do.